### DISCUSSION TITLE
Dynamically updated snippet preview text

### DISCUSSION THREAD

**Discussion by ribru17** [L3MON4D3/LuaSnip#1031] [Q&A]
Hello and thanks for the wonderful plugin! I had a quick question: I followed the wiki example to get a snippet for a markdown table of variable size: 
```lua
  s({ trig = 'tbl(%d+)x(%d+)', regTrig = true }, {
    d(1, function(_, snip)
      local nodes = {}
      local i_counter = 0
      local hlines = ''
      for _ = 1, snip.captures[2] do
        i_counter = i_counter + 1
        table.insert(nodes, t('| '))
        table.insert(nodes, i(i_counter, 'Column_' .. i_counter))
        table.insert(nodes, t(' '))
        hlines = hlines .. '| -------- '
      end
      table.insert(nodes, t { '|', '' })
      hlines = hlines .. '|'
      table.insert(nodes, t { hlines, '' })
      for c_i = 1, snip.captures[1] do
        for c_j = 1, snip.captures[2] do
          i_counter = i_counter + 1
          table.insert(nodes, t('| '))
          table.insert(nodes, i(i_counter, 'Cell_' .. c_i .. '_' .. c_j))
          table.insert(nodes, t(' '))
        end
        table.insert(nodes, t { '|', '' })
      end
      return sn(nil, nodes)
    end),
  }),
```
but when expanding the snippet it looks like this:
![luasnipquestion](https://github.com/L3MON4D3/LuaSnip/assets/55766287/d5dcc23a-9fe0-451f-a734-0680efa424f8)
and I've seen in some videos (and noticed in e.g. the emmet lsp snippets) that there are dynamic snippet previews, i.e. I would like my table snippet preview to e.g. show `tbl3x5~` when i am typing a 3x5 table, and if I type another 5 for example I would like it to show the preview `tbl3x55~`. Is it possible to do this with such a snippet? I have looked everywhere and not had much luck. Thank you!

PS: An example of what I am talking about with a dynamic snippet preview:
![customemmetsnippet](https://github.com/L3MON4D3/LuaSnip/assets/55766287/9117f981-e745-4071-bb47-c6516e2fd3e4)

**Comment by atticus-sullivan** [comment]
I'm also not aware of a way to get the functionality you want. So this is more a feature request, but I think this is more of an issue what `cmp` shows you. Thus [`saadparwaiz1/cmp_luasnip`](https://github.com/saadparwaiz1/cmp_luasnip/) would be the right place to implement this.
Also have a look at https://github.com/saadparwaiz1/cmp_luasnip/issues/6, I think this is closely related.

**Reply by ribru17** [reply]
Ahh thanks so much for the response. I also see that there was an issue raised about this in this repo a while ago that I missed. Thanks for pointing me in the right direction! Going to close this now.
